### PR TITLE
Fix bug 1704038: Upgrade to Django 3.2

### PR DESCRIPTION
**Pull request by PFischbeck**
This upgrades Pontoon to Django 3.2, which means we should be good to go until April 2024 :smile: 

App configs are now automatically discovered, and most of our current app configs were redundant or had wrong names.
The `DEFAULT_AUTO_FIELD` is needed to prevent unwanted migrations in the future, when the default value will change to BigAutoField.

I'll submit a small additional PR to make use of the new `@admin.display` decorator.

### CODE REVIEW

**Comment by PFischbeck** [comment]
I don't understand the CI Docker build failure, locally I can build the Docker container just fine. In particular, `collectstatic` doesn't fail for me.

**Comment by Pike** [comment]
I can reproduce locally, and for me it seems that 

https://github.com/PFischbeck/pontoon/blob/django-3.2/pontoon/tags/__init__.py#L1

is the culprit.

**Review by Pike — APPROVED** [review]

**Comment by mathjazz** [comment]
Thanks for the patch!

I've deployed to stage and noticed /projects is failing:

```
Traceback (most recent call last):
    content = loader.render_to_string(template_name, context, request, using=using)
    raise value.with_traceback(tb)
  File "/app/pontoon/projects/templates/projects/projects.html", line 52, in block "heading"
    {{ HeadingInfo.progress_chart_legend(projects.get_stats_sum()) }}
Traceback (most recent call last):
  File "/app/.heroku/python/lib/python3.8/site-packages/django/core/handlers/exception.py", line 47, in inner
  File "/app/.heroku/python/lib/python3.8/site-packages/django/core/handlers/exception.py", line 49, in inner
  File "/app/.heroku/python/lib/python3.8/site-packages/django/core/handlers/exception.py", line 114, in response_for_exception
  File "/app/.heroku/python/lib/python3.8/site-packages/newrelic/hooks/framework_django.py", line 490, in wrapper
  File "/app/.heroku/python/lib/python3.8/site-packages/newrelic/hooks/framework_django.py", line 626, in wrapper
TypeError: cannot unpack non-iterable function object
  File "/app/.heroku/python/lib/python3.8/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = response or self.get_response(request)
  File "/app/.heroku/python/lib/python3.8/site-packages/newrelic/hooks/framework_django.py", line 1203, in _wrapper
  File "/app/.heroku/python/lib/python3.8/site-packages/django/core/handlers/exception.py", line 49, in inner
    response = response_for_exception(request, exc)
    response = handle_uncaught_exception(request, get_resolver(get_urlconf()), sys.exc_info())
    return _wrapped(*args, **kwargs)
    return wrapped(request, resolver, exc_info)
  File "/app/.heroku/python/lib/python3.8/site-packages/django/core/handlers/exception.py", line 152, in handle_uncaught_exception
    callback = resolver.resolve_error_handler(500)

Traceback (most recent call last):
  File "/app/.heroku/python/lib/python3.8/site-packages/django/utils/deprecation.py", line 117, in __call__
    response = handle_uncaught_exception(request, get_resolver(get_urlconf()), sys.exc_info())
  File "/app/.heroku/python/lib/python3.8/site-packages/newrelic/hooks/framework_django.py", line 490, in wrapper
    return wrapped(request, resolver, exc_info)
    callback = resolver.resolve_error_handler(500)
  File "/app/.heroku/python/lib/python3.8/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
    return wrapped(*args, **kwargs)
    callback = resolver.resolve_error_handler(500)
  File "/app/.heroku/python/lib/python3.8/site-packages/newrelic/hooks/framework_django.py", line 626, in wrapper
TypeError: cannot unpack non-iterable function object

Traceback (most recent call last):
    response = response or self.get_response(request)
    callback, param_dict = wrapped(*args, **kwargs)
    response = get_response(request)
  File "/app/.heroku/python/lib/python3.8/site-packages/django/utils/deprecation.py", line 117, in __call__
    response = response_for_exception(request, exc)
    return wrapped(request, resolver, exc_info)
  File "/app/.heroku/python/lib/python3.8/site-packages/django/core/handlers/exception.py", line 152, in handle_uncaught_exception
  File "/app/.heroku/python/lib/python3.8/site-packages/newrelic/hooks/framework_django.py", line 626, in wrapper
    callback, param_dict = wrapped(*args, **kwargs)
TypeError: cannot unpack non-iterable function object
```

**Comment by Pike** [comment]
I bet this is an old `newrelic` bug, we're a few years behind on updates there.

**Comment by PFischbeck** [comment]
Turns out this does come from the Django update, and is caused by our own code:
> Instantiating an abstract model now raises TypeError.

We instantiate the abstract model `AggregatedStats`. I'll add a regression test and fix it. Maybe there's also a follow-up bug in `newrelic` when handling/passing on the root cause.

**Comment by PFischbeck** [comment]
@mathjazz I've pushed a fix, the same error also happened for `teams/`. It's now covered by tests and should hopefully work in staging.

(I also did some drive-by improvement of test fixture usage in `projects/tests/test_views.py`)

File: pontoon/projects/tests/test_views.py [hunk 1]
```diff
@@ -4,12 +4,17 @@
...
-from pontoon.base.models import Project
-from pontoon.base.tests import (
-    ProjectFactory,
-    ResourceFactory,
-    TranslationFactory,
-)
+from pontoon.base.tests import TranslationFactory
+
+
+@pytest.mark.django_db
```
Comment (Pike) [inline pontoon/projects/tests/test_views.py:10]: I'd give this a https://pytest-cov.readthedocs.io/en/latest/markers-fixtures.html#no-cover, `@pytest.mark.no_cover`.

There's not enough assertions in the test, IMHO, to warrant counting this as test coverage. Doesn't make this a bad test, just not claim coverage for it.

File: pontoon/teams/tests/test_views.py [hunk 2]
```diff
@@ -22,6 +22,16 @@ def managers():
     return _get_sorted_users()
 
 
+@pytest.mark.django_db
```
Comment (Pike) [inline pontoon/teams/tests/test_views.py:25]: `@pytest.mark.no_cover` here, too, for the same reasons as above.

**Review by Pike — CHANGES_REQUESTED** [review]
This looks good, just one more thing about test coverage (or the claim of it).

File: pontoon/projects/tests/test_views.py (hunk 1 above)
Comment (PFischbeck) [reply pontoon/projects/tests/test_views.py:11]: Good idea, thanks!

**Review by mathjazz — APPROVED** [review]
Verified on stage.

Nice work!

**Pull request merged**